tmt/steps/report/reportportal.py

```diff
@@ -1,152 +1,221 @@
+            launch_uuid = yaml_to_dict(response.text).get("id")
+            assert launch_uuid is not None
+            self.info("uuid", launch_uuid, "yellow", shift=1)
+
+            # For each test
+            for result in self.step.plan.execute.results():
+                test = [test for test in self.step.plan.discover.tests()
+                        if test.serialnumber == result.serialnumber][0]
+                item_attributes = attributes.copy()
+                if test.contact:
+                    item_attributes.append({"key": "contact", "value": test.contact[0]})
+                env_vars = [{'key': k, 'value': v} for k, v in test.environment.items()
+                            if not k.startswith("TMT_") or tmt_vars]
+
+                #  Create a test item
+                self.info("test", result.name, color="cyan")
+                response = session.post(url=f"{url}/item",
+                                        headers=headers,
+                                        json={"name": result.name,
+                                              "description": test.summary,
+                                              "attributes": item_attributes,
+                                              "parameters": env_vars,
+                                              "codeRef": test.web_link() or None,
+                                              "launchUuid": launch_uuid,
+                                              "type": "step",
+                                              "testCaseId": test.id or None,
+                                              "startTime": timestamp()})
+                self.handle_response(response)
+                item_uuid = yaml_to_dict(response.text).get("id")
+                assert item_uuid is not None
+                self.info("uuid", item_uuid, "yellow", shift=1)
+
+                # For each log
+                for log_path in result.log:
+                    try:
+                        log = self.step.plan.execute.read(log_path)
+                    except tmt.utils.FileError:
+                        continue
+
+                    level = "INFO" if log_path == result.log[0] else "TRACE"
+                    status = self.TMT_TO_RP_RESULT_STATUS[result.result]
+
+                    # Upload logs
+                    response = session.post(url=f"{url}/log/entry",
+                                            headers=headers,
+                                            json={"message": log,
+                                                  "itemUuid": item_uuid,
+                                                  "launchUuid": launch_uuid,
+                                                  "level": level,
+                                                  "time": timestamp()})
+                    self.handle_response(response)
+
+                    # Write out failures
+                    if log_path == result.log[0] and status == "FAILED":
```

Comment:
Consider using `is` instead of `==`: if the idea here is to check whether the current log is the first log in the list, then the `log_path is result.log[0]` would compare those two objects on the object level rather then whether they contain the same path. I'm not 100 % sure whether it could happen, but if `result.log` be `[Path('/foo.txt'), Path('/foo.txt')]`, your test would trigger for both, `is` just for the first one because only the first object in the list would *be* the `log_path`.

Or, `enumerate()` might be also a cleaner choice: `for i, log_path in enumerate(result.log)`, then `if i == 0 and status == "FAILED"` - focuses on the index alone, which is what the check is about: "are we processing the first log on the list?" - `enumerate()` would add the index in a Pythonic way, it's a well-recognized pattern.

Reply:
Thanks for the tip, rewritten to `enumerate()`.